src/Aspire.Hosting/MongoDB/MongoDbBuilderExtensions.cs

```diff
@@ -0,0 +1,92 @@
+/// </summary>
+public static class MongoDbBuilderExtensions
+{
+    private const int DefaultContainerPort = 27017;
+    private const string DefaultPassword = "password";
+    private const string PasswordEnvVarName = "MONGO_INITDB_ROOT_PASSWORD";
+
+    /// <summary>
+    /// Adds a MongoDB container to the application model. The default image is "mongo" and the tag is "latest".
+    /// </summary>
+    /// <returns></returns>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The host port for MongoDB.</param>
+    /// <param name="password">The password for the MongoDB root user. Defaults to a 'password' password.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{MongoDbContainerResource}"/>.</returns>
+    public static IResourceBuilder<MongoDbContainerResource> AddMongoDbContainer(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        int port = DefaultContainerPort,
+        string password = DefaultPassword)
+    {
+        var mongoDbContainer = new MongoDbContainerResource(name, password);
+
+        return builder
+            .AddResource(mongoDbContainer)
+            .WithAnnotation(new ManifestPublishingCallbackAnnotation(WriteMongoDbContainerToManifest))
+            .WithAnnotation(new ServiceBindingAnnotation(ProtocolType.Tcp, port: port, containerPort: DefaultContainerPort)) // Internal port is always 27017.
+            .WithAnnotation(new ContainerImageAnnotation { Image = "mongo", Tag = "latest" })
+            .WithEnvironment(PasswordEnvVarName, () => mongoDbContainer.Password);
+    }
+
+    /// <summary>
+    /// Adds a MongoDB connection to the application model. Connection strings can also be read from the connection string section of the configuration using the name of the resource.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="connectionString">The MongoDB connection string (optional).</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{MongoDbConnectionResource}"/>.</returns>
+    public static IResourceBuilder<MongoDbConnectionResource> AddMongoDbConnection(this IDistributedApplicationBuilder builder, string name, string? connectionString = null)
+    {
+        var mongoDbConnection = new MongoDbConnectionResource(name, connectionString);
+
+        return builder
+            .AddResource(mongoDbConnection)
+            .WithAnnotation(new ManifestPublishingCallbackAnnotation((json) => WriteMongoDbConnectionToManifest(json, mongoDbConnection)));
```

Comment:
`.WithManifestPublishingCallback(...)`

Reply:
I've changed to use the new method, please review.